src/modsim2/attack/adversarial_images.py

```diff
@@ -35,13 +38,15 @@ def select_best_attack(
 
     # Best attack selection loop
     advs_images = []
+    advs_success = torch.zeros(num_epsilon, device=success.device)
     num_attack_images = len(images[0])
     for i in range(num_attack_images):
         for j in range(num_epsilon):
             if success[j][i] or j == (num_epsilon - 1):
```

Comment:
just to check my understanding, the `or j == (num_epsilon -1)` is so that you always add an image to `advs_images`?

could you add some comments to this for block to explain what you're doing?

Reply:
That's correct, will add some comments